src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -558,80 +559,85 @@ public static LockFileTargetLibrary CreateLockFileTargetProject(
+                                orderedCriteria,
+                                contentItems,
+                                targetGraph.Conventions.Patterns.CompileRefAssemblies,
+                                targetGraph.Conventions.Patterns.CompileLibAssemblies);
 
-                    // Create an ordered list of selection criteria. Each will be applied, if the result is empty
-                    // fallback frameworks from "imports" will be tried.
-                    // These are only used for framework/RID combinations where content model handles everything.
-                    var orderedCriteria = CreateCriteria(targetGraph.Conventions, targetGraph.Framework, targetGraph.RuntimeIdentifier);
+                            projectLib.CompileTimeAssemblies = ConvertToProjectPaths(fileLookup, projectDir, compileGroup);
 
-                    // Compile
-                    // ref takes precedence over lib
-                    var compileGroup = GetLockFileItems(
-                        orderedCriteria,
-                        contentItems,
-                        targetGraph.Conventions.Patterns.CompileRefAssemblies,
-                        targetGraph.Conventions.Patterns.CompileLibAssemblies);
+                            // Runtime
+                            var runtimeGroup = GetLockFileItems(
+                                orderedCriteria,
+                                contentItems,
+                                targetGraph.Conventions.Patterns.RuntimeAssemblies);
 
-                    projectLib.CompileTimeAssemblies = ConvertToProjectPaths(fileLookup, projectDir, compileGroup);
+                            projectLib.RuntimeAssemblies = ConvertToProjectPaths(fileLookup, projectDir, runtimeGroup);
+                        }
+                        else
+                        {
+                            // If the project did not provide a list of assets, add in default ones.
+                            contentItems.Load([libAnyPath]);
 
-                    // Runtime
-                    var runtimeGroup = GetLockFileItems(
-                        orderedCriteria,
-                        contentItems,
-                        targetGraph.Conventions.Patterns.RuntimeAssemblies);
+                            // When there's only lib assets, compile and runtime groups are always equivalent.
+                            var compileGroup = GetLockFileItems(
+                                orderedCriteria,
+                                contentItems,
+                                targetGraph.Conventions.Patterns.CompileLibAssemblies);
 
-                    projectLib.RuntimeAssemblies = ConvertToProjectPaths(fileLookup, projectDir, runtimeGroup);
+                            if (compileGroup.Count > 0)
+                            {
+                                string relativePath = PathUtility.GetPathWithForwardSlashes(Path.Combine("bin", "placeholder", $"{localMatch.Library.Name}.dll"));
```

Comment:
This seems like it could be made to allocate less
```suggestion
                                string relativePath = "bin/placeholder/" + localMatch.Library.Name + ".dll");
```